packages/stream-model-instance-handler/src/__tests__/model-instance-document-handler.test.ts

```diff
@@ -739,6 +803,112 @@ describe('ModelInstanceDocumentHandler', () => {
     expect(state).toMatchSnapshot()
   })
 
+  it('MIDs with SET account relation validate signed commit fields', async () => {
+    const genesisCommit = (await ModelInstanceDocument._makeGenesis(
+      context.signer,
+      null,
+      { ...DETERMINISTIC_METADATA, model: FAKE_MODEL_SET_ID },
+      ['foo', 'bar']
+    )) as GenesisCommit
+    await ipfs.dag.put(genesisCommit, FAKE_CID_1)
+
+    // apply genesis
+    const genesisCommitData = {
+      cid: FAKE_CID_1,
+      type: CommitType.GENESIS,
+      commit: genesisCommit,
+    }
+    let state = await handler.applyCommit(genesisCommitData, context)
+
+    const state$ = TestUtils.runningState(state)
+    const doc = new ModelInstanceDocument(state$, context)
+
+    const signedCommitFail = (await ModelInstanceDocument.makeUpdateCommit(
+      context.signer,
+      doc.commitId,
+      doc.content,
+      { one: 'foo', two: 'baz', myData: 2 }
+    )) as SignedCommitContainer
+    await ipfs.dag.put(signedCommitFail, FAKE_CID_2)
+
+    const payloadFail = dagCBOR.decode(signedCommitFail.linkedBlock)
+    await ipfs.dag.put(payloadFail, signedCommitFail.jws.link)
+
+    const signedCommitDataFail = {
+      cid: FAKE_CID_2,
+      type: CommitType.SIGNED,
+      commit: payloadFail,
+      envelope: signedCommitFail.jws,
+    }
+    await expect(handler.applyCommit(signedCommitDataFail, context, state)).rejects.toThrow(
+      'Unique content fields value does not match metadata. If you are trying to change the value of these fields, this is causing this error: these fields values are not mutable.'
```

Comment:
i would like the error to include the field name in question this would be helpful to developers

Reply:
I guess we could extract the information from the existing `unique` field to detect what field(s) are affected, but hopefully it's not an error that would happen much, as ComposeDB would catch the issue before doing the mutation.